### ISSUE TITLE
Add disableBackdrop to BottomSheet Options

### ISSUE THREAD

**Issue by chriseyhorn** [angular/material#2846]
Allow for the ability to disable the backdrop overlay creation when opening the bottom sheet.

**Comment by chriseyhorn** [comment]
```javascript
/* @ngInject */
  function bottomSheetDefaults($animate, $mdConstant, $mdUtil, $timeout, $compile, $mdTheming, $mdBottomSheet, $rootElement, $mdGesture) {
    var backdrop;

    return {
      themable: true,
      targetEvent: null,
      onShow: onShow,
      onRemove: onRemove,
      escapeToClose: true,
      disableParentScroll: true,
      disableBackdrop: false
    };


    function onShow(scope, element, options) {

      element = $mdUtil.extractElementByName(element, 'md-bottom-sheet');

      if (!options.disableBackdrop) {
        // Add a backdrop that will close on click
        backdrop = $compile('<md-backdrop class="md-opaque md-bottom-sheet-backdrop">')(scope);
        backdrop.on('click', function() {
          $timeout($mdBottomSheet.cancel);
        });
        $mdTheming.inherit(backdrop, options.parent);

        $animate.enter(backdrop, options.parent, null);
      }

      var bottomSheet = new BottomSheet(element, options.parent);
      options.bottomSheet = bottomSheet;

      // Give up focus on calling item
      options.targetEvent && angular.element(options.targetEvent.target).blur();
      $mdTheming.inherit(bottomSheet.element, options.parent);

      if (options.disableParentScroll) {
        options.lastOverflow = options.parent.css('overflow');
        options.parent.css('overflow', 'hidden');
      }

      return $animate.enter(bottomSheet.element, options.parent)
        .then(function() {
          var focusable = angular.element(
            element[0].querySelector('button') ||
            element[0].querySelector('a') ||
            element[0].querySelector('[ng-click]')
          );
          focusable.focus();

          if (options.escapeToClose) {
            options.rootElementKeyupCallback = function(e) {
              if (e.keyCode === $mdConstant.KEY_CODE.ESCAPE) {
                $timeout($mdBottomSheet.cancel);
              }
            };
            $rootElement.on('keyup', options.rootElementKeyupCallback);
          }
        });

    }

    function onRemove(scope, element, options) {

      var bottomSheet = options.bottomSheet;

      if (!options.disableBackdrop) {
        $animate.leave(backdrop);
      }
      return $animate.leave(bottomSheet.element).then(function() {
        if (options.disableParentScroll) {
          options.parent.css('overflow', options.lastOverflow);
          delete options.lastOverflow;
        }

        bottomSheet.cleanup();

        // Restore focus
        options.targetEvent && angular.element(options.targetEvent.target).focus();
      });
    }
```

**Comment by marcysutton** [comment]
Why don't you submit a PR?

**Comment by plato-cambrian** [comment]
I would like to disable the backdrop on my md-select as well, could a `md-no-backdrop` directive work?

I have a list of sounds in the select, and when you click the option I play a preview of the sound and set my data model.

Because the common use case will be 'try all the sounds' I am stopping propagation of the option click event to keep the select open.

Unfortunately, because of the backdrop, once the user has selected a sound they have to click 'Save Settings' twice, once to close the backdrop and once to click the save button.

**Comment by sergkr** [comment]
I'd like to have the ability to disable the backdrop on the sidenav component as well.

**Comment by mackelito** [comment]
+1

**Comment by DevVersion** [comment]
There is a dupe of this issue, see #6473. This feature is also implemented for `bottomsheet` and `sidenav` in #6434.
